shared/src/lib.rs

```diff
@@ -142,6 +142,34 @@ pub struct State<Phase: EDHOCState> {
     pub th_3: BytesHashLen,
 }
 
+#[derive(Debug)]
+pub struct PartialMessage1 {
+    pub c_i: u8,                    // connection identifier chosen by the initiator
+    pub x_or_y: BytesP256ElemLen,   // ephemeral private key of myself
+    pub gx_or_gy: BytesP256ElemLen, // g_y or g_x, ephemeral public key of myself
+    pub suites_i: BytesSuites,
+    pub suites_i_len: usize,
+}
+
+#[derive(Debug)]
+pub struct WaitMessage2New {
```

Comment:
AIU WaitMessage2New replaces the State<WaitMessage2>. That's a welcome replacement -- can the WaitMessage2 be removed now already?

Reply:
Yep, it is WIP and thus a bit of a mess for now; in the end I will remove all the old stuff.